src/backend/distributed/connection/remote_commands.c

```diff
@@ -994,14 +994,20 @@ BuildWaitEventSet(MultiConnection **allConnections, int totalConnectionCount,
 	int pendingConnectionCount = totalConnectionCount - pendingConnectionsStartIndex;
 	WaitEventSet *waitEventSet = NULL;
 	int connectionIndex = 0;
+	const int MAX_WAIT_EVENT_COUNT = 50;
```

Comment:
This should probably be a function of `FD_SETSIZE` instead of constant 50 everywhere.

Reply:
Oh yeah, good point